numpy/core/src/multiarray/descriptor.c

```diff
@@ -90,31 +90,51 @@ _arraydescr_from_ctypes_type(PyTypeObject *type)
  * and it can be converted to a dtype object.
  *
  * Returns a new reference to a dtype object, or NULL
- * if this is not possible. When it returns NULL, it does
- * not set a Python exception.
+ * if this is not possible.
+ * When the return value is true, the dtype attribute should have been used
+ * and parsed. Currently the only failure mode for a 1 return is a
```

Comment:
Returning `1` for an error is unusual. Can you return `-1` instead, and use the standard `< 0` check at the call site?

Reply:
Yeah, did that because it actually meant "using the dtype attribute" and not necessarily error, but maybe nicer in any case.

Reply:
Ah, I get it now. Perhaps just a better docstring - in particular, line 92 is now wrong.  Something like:
```
Returns 1 if getting a dtype was attempted, otherwise 0.
If an attempt was made, `newdescr` will contain the result, or NULL if an error occured and should be propagated.
```
You could also consider returning `PyNone`, but then you have to remember to reference count it which is a bit of a waste of time.

Reply:
Yeah, PyNone may be a neat idea, it is just one "DECREF on error" in a sense, so maybe nicer than this dance.

About splitting off: Yeah, thought it would be better as well probably, so planning to do that, if I don't get to it today, might be a few days.